libs/sdk-core/src/breez_services.rs

```diff
@@ -1140,32 +1232,46 @@ impl BreezServices {
     /// If not or no LSP is selected, it selects the first LSP in [`list_lsps`].
     async fn connect_lsp_peer(&self, node_pubkey: String) -> SdkResult<()> {
         let lsps = self.lsp_api.list_lsps(node_pubkey).await?;
-        if let Some(lsp) = self
+        let lsp = match self
             .persister
             .get_lsp_id()?
-            .and_then(|lsp_id| lsps.clone().into_iter().find(|lsp| lsp.id == lsp_id))
-            .or_else(|| lsps.first().cloned())
+            .and_then(|lsp_id| lsps.iter().find(|lsp| lsp.id == lsp_id))
+            .or_else(|| lsps.first())
         {
-            self.persister.set_lsp_id(lsp.id)?;
-            if let Ok(node_state) = self.node_info() {
-                let node_id = lsp.pubkey;
-                let address = lsp.host;
-                let lsp_connected = node_state
-                    .connected_peers
-                    .iter()
-                    .any(|e| e == node_id.as_str());
-                if !lsp_connected {
-                    debug!("connecting to lsp {}@{}", node_id.clone(), address.clone());
-                    self.node_api
-                        .connect_peer(node_id.clone(), address.clone())
-                        .await
-                        .map_err(|e| SdkError::ServiceConnectivity {
-                            err: format!("(LSP: {node_id}) Failed to connect: {e}"),
-                        })?;
-                }
-                debug!("connected to lsp {node_id}@{address}");
-            }
+            Some(lsp) => lsp.clone(),
+            None => return Ok(()),
+        };
+
+        self.persister.set_lsp_id(lsp.id)?;
+        self.persister.set_lsp_pubkey(lsp.pubkey.clone())?;
+        let mut node_state = match self.node_info() {
+            Ok(node_state) => node_state,
+            Err(_) => return Ok(()),
+        };
+
+        let node_id = lsp.pubkey;
+        let address = lsp.host;
+        let lsp_connected = node_state
+            .connected_peers
+            .iter()
+            .any(|peer| peer.id == node_id);
+        if !lsp_connected {
+            debug!("connecting to lsp {}@{}", node_id.clone(), address.clone());
+            let features = self
+                .node_api
+                .connect_peer(node_id.clone(), address.clone())
+                .await
+                .map_err(|e| SdkError::ServiceConnectivity {
+                    err: format!("(LSP: {node_id}) Failed to connect: {e}"),
+                })?;
+            node_state.connected_peers.push(ConnectedPeer {
+                id: node_id.clone(),
+                features,
+            });
+            self.persister.set_node_state(&node_state)?;
```

Comment:
I'm wondering if updating the node_state's connected peers could done in `node_api.connect_peer()`, but AFAIK this is the only time its called